sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -218,6 +218,44 @@ struct ImageDeleter {
   }
 };
 
+static bool is_compiler_available(const context &C,
+                                  RT::PiDeviceBinaryType Format) {
+  // No compilation needed for native format
+  if (Format == PI_DEVICE_BINARY_TYPE_NATIVE)
+    return true;
+
+  // All other formats need a compiler to be available
+  auto P = C.get_platform();
+  bool found_compiler = false;
+  for (const auto &D : C.get_devices()) {
+    if (D.get_info<info::device::is_compiler_available>()) {
+      found_compiler = true;
+      break;
+    }
+  }
+  if (!found_compiler)
+    return false;
+
+  // SPIR-V additionally needs to be able to compile IL
+  if (Format == PI_DEVICE_BINARY_TYPE_SPIRV) {
```

Comment:
An early return could improve readability:
```
if (Format != PI_DEVICE_BINARY_TYPE_SPIRV)
  return false;
// Further code
```

Reply:
I think it has to be return true here, not false, as we have found a compiler